backend/vecsim_app/load_data.py

```diff
@@ -1,32 +1,37 @@
 #!/usr/bin/env python3
-import typing as t
 import asyncio
-import numpy as np
 import pickle
-import redis.asyncio as redis
+import typing as t
 
+import numpy as np
+import redis.asyncio as redis
 from redis.commands.search.field import TagField
 from vecsim_app import config
 from vecsim_app.models import Paper
 from vecsim_app.search_index import SearchIndex
 
 
 def read_paper_df() -> t.List:
-    with open(config.DATA_LOCATION + "/arxiv_embeddings_10000.pkl", "rb") as f:
+    path = config.DATA_LOCATION + "/embeddings/arxiv_embeddings_400000.pkl"
+    print(f"Loading data from   : {path}")
+    with open(path, "rb") as f:
         df = pickle.load(f)
+    print(f"Loaded {len(df)} items")
     return df
 
+
 async def gather_with_concurrency(n, redis_conn, *papers):
     semaphore = asyncio.Semaphore(n)
+
     async def load_paper(paper):
         async with semaphore:
-            vector = paper.pop('vector')
-            paper['paper_id'] = paper.pop('id')
+            vector = paper.pop("vector")
+            paper["paper_id"] = paper.pop("id")
             # TODO - we need to be able to use other separators
-            paper['categories'] = paper['categories'].replace(",", "|")
+            paper["categories"] = paper["categories"].replace(",", "|")
             p = Paper(**paper)
             # save model TODO -- combine these two objects eventually
-            await p.save()
+            await p.save(redis_conn)
```

Comment:
note: that was a bug, I remember having problems when working on remote machine in Saturn cloud with remote Redis instance